### ISSUE TITLE
[beta] Backspace key (possibly other) isn't accepted by client

### ISSUE THREAD

**Issue by guss77** [symless/synergy#5460]
I'm running Synergy 1.8.2-beta-8da57cd on Ubuntu 16.04

When pressing the backspace key on the server, the client has this in the logs:

~~~
[2016-07-12T13:22:42] DEBUG1: recv key down id=0x0000ef08, mask=0x0000, button=0x0016
[2016-07-12T13:22:42] DEBUG1: mapKey ef08 (61192) with mask 0000, start state: 2000
[2016-07-12T13:22:42] DEBUG1: find best:  2000 0000
[2016-07-12T13:22:42] DEBUG1: best key index 2 of 2 (exact)
[2016-07-12T13:22:42] DEBUG1: found key in group 0
[2016-07-12T13:22:42] DEBUG1: state: 2000,0001,0001
[2016-07-12T13:22:42] DEBUG1: flip: 0001 (2000 vs 0001 in 0001 - 0000)
[2016-07-12T13:22:42] DEBUG1: modifier 0001 modified by 0004
[2016-07-12T13:22:42] DEBUG1: unable to match modifier state (0001,0001) for key 22
[2016-07-12T13:22:42] DEBUG1: can't map key
[2016-07-12T13:22:42] DEBUG1: recv key up id=0x0000ef08, mask=0x0000, button=0x0016
~~~

and the backspace key is not sent to the application on the client.

The following may be related:
The client is using a bi-lingual keyboard setup with two keyboard layouts - this is what `setxkbmap -print` shows:

~~~
xkb_keymap {
        xkb_keycodes  { include "evdev+aliases(qwerty)" };
        xkb_types     { include "complete"      };
        xkb_compat    { include "complete"      };
        xkb_symbols   { include "pc+us+il(lyx):2+inet(evdev)+group(alt_shift_toggle)"   };
        xkb_geometry  { include "pc(pc101)"     };
};
~~~
while the server is using a standard `us` layout.

**Comment by guss77** [comment]
The TAB key also misbehaves

**Comment by XinyuHou** [comment]
This is a side effect from the fix of #2765. We are investigating it. Please leave further comments in #2765
